lxc/utils.go

```diff
@@ -239,6 +243,178 @@ func getConfig(args ...string) (map[string]string, error) {
 	return values, nil
 }
 
+// stringToTimeHookFunc is a custom decoding hook that converts string values to time.Time using the given layout.
+func stringToTimeHookFunc(layout string) mapstructure.DecodeHookFuncType {
+	return func(from reflect.Type, to reflect.Type, data interface{}) (interface{}, error) {
+		if from.Kind() == reflect.String && to == reflect.TypeOf(time.Time{}) {
+			strValue := data.(string)
+			t, err := time.Parse(layout, strValue)
+			if err != nil {
+				return nil, err
+			}
+
+			return t, nil
+		}
+
+		return data, nil
+	}
+}
+
+// stringToBoolHookFunc is a custom decoding hook that converts string values to bool.
+func stringToBoolHookFunc() mapstructure.DecodeHookFunc {
+	return func(f reflect.Kind, t reflect.Kind, data interface{}) (interface{}, error) {
+		if f != reflect.String || t != reflect.Bool {
+			return data, nil
+		}
+
+		str := data.(string)
+		str = strings.ToLower(str)
+		switch str {
+		case "1", "t", "true":
+			return true, nil
+		default:
+			return false, nil
+		}
+	}
+}
+
+// stringToIntHookFunc is a custom decoding hook that converts string values to int.
+func stringToIntHookFunc() mapstructure.DecodeHookFunc {
+	return func(f reflect.Kind, t reflect.Kind, data interface{}) (interface{}, error) {
+		if f != reflect.String || (t != reflect.Int && t != reflect.Int8 && t != reflect.Int16 && t != reflect.Int32 && t != reflect.Int64) {
+			return data, nil
+		}
+
+		str := data.(string)
+		value, err := strconv.Atoi(str)
+		if err != nil {
+			return data, err
+		}
+
+		return value, nil
+	}
+}
+
+// stringToFloatHookFunc is a custom decoding hook that converts string values to float.
+func stringToFloatHookFunc() mapstructure.DecodeHookFunc {
+	return func(f reflect.Kind, t reflect.Kind, data interface{}) (interface{}, error) {
+		if f != reflect.String || (t != reflect.Float32 && t != reflect.Float64) {
+			return data, nil
+		}
+
+		str := data.(string)
+		value, err := strconv.ParseFloat(str, 64)
+		if err != nil {
+			return data, err
+		}
+
+		return value, nil
+	}
+}
+
+// getFieldByJsonTag gets the value of a struct field by its JSON tag.
+func getFieldByJsonTag(obj any, tag string) (any, error) {
+	var res any
+	ok := false
+	v := reflect.ValueOf(obj).Elem()
+	for i := 0; i < v.NumField(); i++ {
+		jsonTag := v.Type().Field(i).Tag.Get("json")
+
+		// Ignore any options that might be specified after a comma in the tag.
+		commaIdx := strings.Index(jsonTag, ",")
+		if commaIdx > 0 {
+			jsonTag = jsonTag[:commaIdx]
+		}
+
+		if strings.EqualFold(jsonTag, tag) {
+			res = v.Field(i).Interface()
+			ok = true
+		}
+	}
+
+	if !ok {
+		return nil, fmt.Errorf("The property with tag %q does not exist", tag)
+	}
+
+	return res, nil
+}
+
+// setFieldByJsonTag sets the value of a struct field by its JSON tag.
```

Comment:
These 3 functions below seem to be good candidates for having some unit tests as they are self-contained, especially as they are dealing with different data types using reflect.